xcode/Subconscious/Shared/Services/OnThisDayGeist.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  OnThisDayGeist.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 30/9/2022.
+//
+
+import Foundation
+
+
+enum OnThisDayVariant: CaseIterable {
+    case inTheLastDay
+    case inTheLastWeek
+    case aYearAgo
+    case sixMonthsAgo
+    case aMonthAgo
+    
+    static func random() -> OnThisDayVariant {
+        // https://stackoverflow.com/questions/63189870/return-a-random-case-from-an-enum-in-swift
+        return allCases.randomElement()!
+    }
+}
+
+struct OnThisDayGeist: Geist {
+    private let database: DatabaseService
+
+    init(database: DatabaseService) {
+        self.database = database
+    }
+    
+    func addTo(date: Date, component: Calendar.Component, value: Int) -> Date? {
+        guard let d = Calendar.current.date(byAdding: component, value: value, to: date) else {
+            return nil
+        }
+        
+        return d
+    }
+    
+    func dateFromVariant(variant: OnThisDayVariant) -> (Date, Date)? {
```

Comment:
I suggest creating a struct called DateRange. Reason: I've had issues in the past with the Swift compiler crashing because it gets confused trying to track anonymous tuple lifetimes.

```
struct DateRange: Hashable {
    var beginning: Date
    var end: Date
}
```

Reply:
Done and done :)